### PR TITLE
Always show snippets should not exclude non-snippet items

### PR DESCRIPTION
**Pull request by rchande**
**Customer scenario**

The customer sets the VB option to always show snippets in the completion list, which mimics the C# behavior. Now the completion list contains *only* snippets.

**Bugs this fixes:**

https://github.com/dotnet/roslyn/issues/17951

**Workarounds, if any**

This option is essentially unusable in VB.

**Risk**

Extremely low. This only affects snippets.

**Performance impact**

None

**Is this a regression from a previous update?**

No, but only because we didn't offer this option before.

**Root cause analysis:**

This option was not tested in VB when the same set of completion options were made available for both langauges.

**How was the bug found?**

Dogfooding.

### CODE REVIEW

**Comment by rchande** [comment]
Tag @dotnet/roslyn-ide for review

File: src/VisualStudio/VisualBasic/Impl/Snippets/SnippetCompletionProvider.vb [hunk 1]
```diff
@@ -52,10 +52,14 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Snippets
             Dim syntaxFacts = document.GetLanguageService(Of ISyntaxFactsService)()
             Dim isPossibleTupleContext = syntaxFacts.IsPossibleTupleContext(syntaxTree, position, cancellationToken)
 
-            context.IsExclusive = True
+            context.IsExclusive = ShouldBeExclusive(context.Options)
             context.AddItems(CreateCompletionItems(snippets, isPossibleTupleContext))
         End Function
 
+        Private Function ShouldBeExclusive(options As OptionSet) As Boolean
+            Return not options.GetOption(CompletionOptions.SnippetsBehavior, LanguageNames.VisualBasic) = SnippetsRule.AlwaysInclude
```
Comment (dpoeschl) [inline src/VisualStudio/VisualBasic/Impl/Snippets/SnippetCompletionProvider.vb]: Why not check in the positive against `SnippetsRule.IncludeAfterTypingIdentifierQuestionTab`, which is the one case where it should be exclusive (I think, right?). Would make this more readable.

Comment (Pilchie) [inline src/VisualStudio/VisualBasic/Impl/Snippets/SnippetCompletionProvider.vb]: Agreed.

**Comment by rchande** [comment]
Tag @MattGertz for ask mode approval.

**Comment by MattGertz** [comment]
Approved pending tests.

**Comment by rchande** [comment]
retest this please

**Comment by rchande** [comment]
retest windows_debug_unit32_prtest

**Comment by rchande** [comment]
retest windows_debug_unit32_prtest

**Comment by rchande** [comment]
Failing debug unit tests are in CSharp Move Type tests. They pass on my machine and shouldn't be affected by the change in this PR (which changes VB). @dotnet/roslyn-infrastructure are there issues with the move type tests?

**Comment by rchande** [comment]
VSI failures also appear unrelated to the change.

**Comment by CyrusNajmabadi** [comment]
You can ignore the MoveType tests.  I found and fixed the flakeyness there in the test suite.

**Pull request merged**